### PR TITLE
Fix linking Boost 1.70.0

### PR DESCRIPTION
**Pull request by tobiasleibner**

### CODE REVIEW

File: cmake/modules/DuneXtCommonMacros.cmake [hunk 1]
```diff
@@ -27,6 +27,9 @@ foreach(_boost_lib ${DS_REQUIRED_BOOST_LIBS})
   set(_BOOST_LIB "")
   string(TOUPPER "${_boost_lib}" _BOOST_LIB)
   dune_register_package_flags(LIBRARIES ${Boost_${_BOOST_LIB}_LIBRARY})
+  if (TARGET Boost::${_boost_lib})
```
Comment (renefritze) [inline cmake/modules/DuneXtCommonMacros.cmake]: Is boost 1.70 building with cmake and exporting its targets? :open_mouth:

Comment (tobiasleibner) [reply cmake/modules/DuneXtCommonMacros.cmake]: Yes and ``Boost_*_LIBRARY`` is not defined anymore in that case... But I am getting weird errors from dune-testtools about a missing ``Boost::atomic`` target now, so this should not be merged yet.

Comment (tobiasleibner) [reply cmake/modules/DuneXtCommonMacros.cmake]: Okay, so apparently with these changes it is only working with 1.70.0 due to [this](https://gitlab.kitware.com/cmake/cmake/issues/18619). The suggested fix is to also require the boost atomic library.

Comment (renefritze) [reply cmake/modules/DuneXtCommonMacros.cmake]: > The suggested fix is to also require the boost atomic library.

I see no problem doing that.

**Comment by tobiasleibner** [comment]
We now also require ``Boost.Atomic`` which is needed by the ``Boost.Thread`` library, see [this issue](https://gitlab.kitware.com/cmake/cmake/issues/18619), @dune-community/dune-xt-devs . 

In addition, I removed the vector-valued version of ``clamp`` which is unused, and changed the scalar ``clamp`` signature to conform to ``C++17``'s ``std::clamp``. Without this change, the tests where not compiling with ``-std=c++17``.

**Pull request merged**